### ISSUE TITLE
Steam can’t capture sound when using speexrate_medium or speexrate_best

### ISSUE THREAD

**Issue by vfjpl** [ValveSoftware/steam-for-linux#4547]
#### Your system information

* Steam client version:
* Distribution (e.g. Ubuntu): Manjaro
* Opted into Steam client beta?: [Yes/No] No
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:

When using alsa with defaults.pcm.rate_converter "speexrate_best", steam can't capture from dsnoop.

#### Steps for reproducing this issue:

1. Select "Alsa default" from steam sound options
2. Create /etc/asound.conf with default.pcm.rate_converter "speexrate_medium" or "speexrate_best"
3. Steam can't capture dsnoop

Please provide info if this bug is affecting you or not.

**Comment by gdrewb-valve** [comment]
It's not clear that this is a Steam problem, but it may be.

**Comment by vfjpl** [comment]
Is it Steam related problem? If not can you report it somwhere else? If not, where I can do it?

**Comment by gdrewb-valve** [comment]
I don't believe the Steam team has looked yet.  If they have any info they'll post it here, but it's unclear when that'll be.

**Comment by vfjpl** [comment]
Any news?

**Comment by Eriner** [comment]
I am experiencing as well.

**Comment by Eriner** [comment]
Some relevant error logs:

```
ALSA lib dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_�.so
ALSA lib pcm_rate.c:1362:(snd_pcm_rate_open) Cannot find rate converter
AL lib: (EE) ALCcaptureAlsa_open: snd_pcm_hw_params(self->pcmHandle, hp) failed: No such file or directory
AL lib: (EE) UpdateDeviceParams: Failed to set 16000hz, got 48000hz instead
ALSA lib dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_����x/��e_medium.so
ALSA lib pcm_rate.c:1362:(snd_pcm_rate_open) Cannot find rate converter
AL lib: (EE) ALCcaptureAlsa_open: snd_pcm_hw_params(self->pcmHandle, hp) failed: No such file or directory
ALSA lib dlmisc.c:254:(snd1_dlobj_cache_get) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_rate_�I���{��.so
ALSA lib pcm_rate.c:1362:(snd_pcm_rate_open) Cannot find rate converter
AL lib: (EE) ALCcaptureAlsa_open: snd_pcm_hw_params(self->pcmHandle, hp) failed: No such file or directory

```

Yes, the library name is garbled like that. I haven't run it through a hexdump, however my guess is that this was supposed to be `/usr/lib32/alsa-lib/libasound_module_rate_speexrate_medium.so`, as I have that directive set in my alsa configuration. String corruption happening somewhere, can't find shared object because `libasound_module_rate_�I���{��.so` obviously doesn't exist?

Reproduction steps in OP are accurate.
